pyciemss/interfaces.py

```diff
@@ -528,16 +533,204 @@ def wrapped_model():
     return {"inferred_parameters": inferred_parameters, "loss": loss}
 
 
-# # TODO
-# def optimize(
-#     model: CompiledDynamics,
-#     objective_function: ObjectiveFunction,
-#     constraints: Constraints,
-#     optimization_algorithm: OptimizationAlgorithm,
-#     *args,
-#     **kwargs
-# ) -> OptimizationResult:
-#     """
-#     Optimize the objective function subject to the constraints.
-#     """
-#     raise NotImplementedError
+def optimize(
+    model_path_or_json: Union[str, Dict],
+    end_time: float,
+    logging_step_size: float,
+    qoi: Callable,
+    risk_bound: float,
+    static_parameter_interventions: Dict[torch.Tensor, str],
+    objfun: Callable,
+    initial_guess_interventions: List[float],
+    bounds_interventions: List[List[float]],
+    *,
+    solver_method: str = "dopri5",
+    solver_options: Dict[str, Any] = {},
+    start_time: float = 0.0,
+    inferred_parameters: Optional[pyro.nn.PyroModule] = None,
+    n_samples_ouu: int = int(1e2),
+    maxiter: int = 5,
+    maxfeval: int = 25,
+    verbose: bool = False,
+    roundup_decimal: int = 4,
+    postprocess: bool = False,
+) -> Dict:
```

Comment:
Can we make the return type more refined? Even just `Dict[str, Any]` would help a bit.